target/linux/generic/pending-6.12/811-pci_disable_usb_common_quirks.patch

```diff
@@ -20,16 +20,25 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  static struct amd_chipset_info {
  	struct pci_dev	*nb_dev;
  	struct pci_dev	*smbus_dev;
-@@ -590,6 +592,8 @@ bool usb_amd_pt_check_port(struct device
+@@ -590,6 +592,7 @@ bool usb_amd_pt_check_port(struct device
  EXPORT_SYMBOL_GPL(usb_amd_pt_check_port);
  #endif /* CONFIG_USB_PCI_AMD */
  
-+#endif /* CONFIG_PCI_DISABLE_COMMON_QUIRKS */
-+
++#ifdef CONFIG_USB_PCI
  static int usb_asmedia_wait_write(struct pci_dev *pdev)
  {
  	unsigned long retry_count;
-@@ -724,6 +728,10 @@ reset_needed:
+@@ -633,6 +636,9 @@ void usb_asmedia_modifyflowcontrol(struc
+ 	pci_write_config_byte(pdev, ASMT_CONTROL_REG, ASMT_CONTROL_WRITE_BIT);
+ }
+ EXPORT_SYMBOL_GPL(usb_asmedia_modifyflowcontrol);
++#endif /* CONFIG_USB_PCI */
++
++#endif /* !CONFIG_PCI_DISABLE_COMMON_QUIRKS */
```

Comment:
This is still overlapping with the `#ifdef CONFIG_USB_PCI_AMD`  block. The `#ifdef CONFIG_USB_PCI_AMD` is only activated on x86 and maybe longsoon, they are not space constrained we do not need extra handling in our patch for them. 

I think we only need the last block of the patch for this file, the rest is already taken care of upstream.

Reply:
Take a look at my solution: https://github.com/openwrt/openwrt/pull/18720

Reply:
> This is still overlapping with the `#ifdef CONFIG_USB_PCI_AMD` block. The `#ifdef CONFIG_USB_PCI_AMD` is only activated on x86 and maybe longsoon, they are not space constrained we do not need extra handling in our patch for them.

This makes sense. I have removed the modifications about `CONFIG_USB_PCI_AMD`.

> 
> I think we only need the last block of the patch for this file, the rest is already taken care of upstream.

We still need to exclude `usb_asmedia_modifyflowcontrol()` when `CONFIG_PCI_DISABLE_COMMON_QUIRKS` is enabled. It fixes the function redefinition error. On 6.6 kernel, `usb_asmedia_modifyflowcontrol()` is an empty function, too.